tools/mdm/migration/mdmproxy/mdmproxy.go

```diff
@@ -0,0 +1,312 @@
+		http.Error(w, "Unable to read request body", http.StatusUnprocessableEntity)
+		return
+	}
+	// Reset body so that the reverse proxy request includes it
+	r.Body = io.NopCloser(bytes.NewReader(body))
+
+	// Get the UDID from request
+	udid, err := udidFromRequestBody(body)
+	if err != nil {
+		log.Printf("%s %s Failed to get UDID: %v", r.Method, r.URL.String(), err)
+		http.Error(w, err.Error(), http.StatusUnprocessableEntity)
+		return
+	}
+
+	// Migrated UDIDs go to the Fleet server, otherwise requests go to the existing server.
+	if udid != "" && m.isUDIDMigrated(udid) {
+		log.Printf("%s %s (%s) -> Fleet", r.Method, r.URL.String(), udid)
+		m.fleetProxy.ServeHTTP(w, r)
+	} else {
+		log.Printf("%s %s (%s) -> Existing", r.Method, r.URL.String(), udid)
+		m.existingProxy.ServeHTTP(w, r)
+	}
+}
+
+func (m *mdmProxy) handleUpdatePercentage(w http.ResponseWriter, r *http.Request) {
+	if m.token == "" {
+		http.Error(w, "Set auth token to enable remote updates", http.StatusUnauthorized)
+		return
+	}
+	token := r.Header.Get("Authorization")
+	if token == "" {
+		http.Error(w, "Authorization header must be provided", http.StatusUnauthorized)
+		return
+
+	}
+	if token != m.token {
+		http.Error(w, "Authorization header does not match", http.StatusUnauthorized)
+		return
+	}
+
+	body, err := io.ReadAll(r.Body)
+	defer r.Body.Close()
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Failed to read body: %v", err), http.StatusInternalServerError)
+		return
+	}
+	percentage, err := strconv.Atoi(string(body))
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Cannot read body as integer: %v", err), http.StatusUnprocessableEntity)
+		return
+	}
+	if percentage < 0 || percentage > 100 {
+		http.Error(w, "Percentage should be in range (0, 100)", http.StatusBadRequest)
+		return
+	}
+
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	m.migratePercentage = percentage
+
+	msg := fmt.Sprintf("Migrate percentage updated: %v\n", percentage)
+	log.Printf(msg)
+	fmt.Fprintf(w, msg)
+}
+
+func (m *mdmProxy) handleUpdateMigrateUDIDs(w http.ResponseWriter, r *http.Request) {
+	if m.token == "" {
+		http.Error(w, "Set auth token to enable remote updates", http.StatusUnauthorized)
+		return
+	}
+	token := r.Header.Get("Authorization")
+	if token == "" {
+		http.Error(w, "Authorization header must be provided", http.StatusUnauthorized)
+		return
+
+	}
+	if token != m.token {
+		http.Error(w, "Authorization header does not match", http.StatusUnauthorized)
+		return
+	}
+
+	defer r.Body.Close()
+	udids, err := processUDIDs(r.Body)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	m.migrateUDIDs = udids
+
+	msg := fmt.Sprintf("Migrate UDIDs updated: %v\n", udids)
+	log.Printf(msg)
+	fmt.Fprintf(w, msg)
+}
+
+func processUDIDs(in io.Reader) (map[string]struct{}, error) {
+	scanner := bufio.NewScanner(in)
+	scanner.Split(bufio.ScanWords)
```

Comment:
ScanWords appears to operate space-delimited: https://pkg.go.dev/bufio#ScanWords